src/container/srv_container.c

```diff
@@ -4563,6 +4563,10 @@ upgrade_cont_cb(daos_handle_t ih, d_iov_t *key, d_iov_t *val, void *varg)
 		}
 	}
 
+	entry = daos_prop_entry_get(prop, DAOS_PROP_CO_OBJ_VERSION);
+	D_ASSERT(entry != NULL);
+	entry->dpe_val = DS_POOL_OBJ_VERSION;
+	entry->dpe_flags &= ~DAOS_PROP_ENTRY_NOT_SET;
```

Comment:
@wangshilong, could you help review this PR, please? Also, do you know if we need to do this (i.e., to update `prop`) for the other DB updates in this function?

Reply:
I don't think we need do this, this is only needed when server returning prop to clients, so client know even this prop has value, but it is missed.

Reply:
@wangshilong I was confused what's the purpose of those code either.
But it actually changed the prop's OBJ_VERSION entry and in the below code -
4703 >       >       >       rc = cont_iv_prop_update(ap->svc->cs_pool->sp_iv_ns,                        
4704 >       >       >       >       >       >        cont_uuid, prop, true);   
it update the prop to IV prop cache, seems that is necessary? (keep IV prop cache consistent with local RDB).

Reply:
Yep, if other DB updates in this function do not need this, then I'm also confused by why OBJ_VERSION needs this.

Reply:
> @wangshilong I was confused what's the purpose of those code either. But it actually changed the prop's OBJ_VERSION entry and in the below code - 4703 > > > rc = cont_iv_prop_update(ap->svc->cs_pool->sp_iv_ns, 4704 > > > > > > cont_uuid, prop, true); it update the prop to IV prop cache, seems that is necessary? (keep IV prop cache consistent with local RDB).

You are right, we need this.

Reply:
Yes, the entry is used to update IV cache.

>if other DB updates in this function do not need this, then I'm also confused by why OBJ_VERSION needs this.

This is upgrade callback. so global pool version and object layout version are supposed to be updated. both for DB and IV cache.